tools/install-powershell.sh

```diff
@@ -0,0 +1,137 @@
+#!/bin/bash
+
+#Companion code for the blog https://cloudywindows.com
+
+#call this code direction from the web with:
+#bash <(wget -O - https://raw.githubusercontent.com/PowerShell/PowerShell/master/tools/install-powershell.sh) <ARGUMENTS>
+#wget -O - https://raw.githubusercontent.com/PowerShell/PowerShell/master/tools/install-powershell.sh | bash -s <ARGUMENTS>
+
+#Usage - if you do not have the ability to run scripts directly from the web, 
+#        pull all files in this repo folder and execute, this script
+#        automatically prefers local copies of sub-scripts
+
+#Completely automated install requires a root account or sudo with a password requirement
+
+#Switches
+#  -includeide - the script is being run headless, do not perform actions that require response from the console
+#  -interactivetesting - requires a human user in front of the machine - loads a script into the ide to test with F5 to ensure the IDE can run scripts
+#  -appimage - does appimage instead of native install
+
+#gitrepo paths are overrideable to run from your own fork or branch for testing or private distribution
+
+VERSION="1.1.0"
+gitreposubpath="PowerShell/PowerShell/master"
+gitreposubpath="darwinjs/PowerShell/feature/install-powershell.sh"
+gitreposcriptroot="https://raw.githubusercontent.com/$gitreposubpath/tools"
+gitscriptname="install-powershell.psh"
+
+echo "Get-PowerShell Core MASTER Installer Version $VERSION"
+echo "Installs PowerShell Core and Optional The Development Environment"
+echo "  Original script is at: $gitreposcriptroot\$gitscriptname"
+
+echo "Arguments used: $*"
+echo ""
+
+# Let's quit on interrupt of subcommands
+trap '
+  trap - INT # restore default INT handler
+  echo "Interrupted"
+  kill -s INT "$$"
+' INT
+
+lowercase(){
+    echo "$1" | sed "y/ABCDEFGHIJKLMNOPQRSTUVWXYZ/abcdefghijklmnopqrstuvwxyz/"
+}
```

Comment:
Ok did some research and it seems like anything in the posix command set should be considered universal and "tr" is on the list: http://pubs.opengroup.org/onlinepubs/9699919799/utilities/contents.html